src/runtime_src/core/edge/drm/zocl/cu_scu.c

```diff
@@ -202,9 +199,42 @@ int xrt_cu_scu_init(struct xrt_cu *xcu)
 void xrt_cu_scu_fini(struct xrt_cu *xcu)
 {
 	struct xrt_cu_scu *core = xcu->core;
+	struct pid *p = NULL;
+	struct task_struct *task = NULL;
+	int ret = 0;
+
+	// Wait for PS Kernel Process to finish
+	p = find_get_pid(core->sc_pid);
+	if(!p) {
+		// Process already gone
+		goto skip_kill;
+	}
+
+	task = pid_task(p, PIDTYPE_PID);
+	if(!task) {
+		DRM_WARN("Failed to get task for pid %d\n",core->sc_pid);
+		put_pid(p);
+		goto skip_kill;
+	}
+
+	if(core->sc_parent_pid != task_ppid_nr(task)) {
+		DRM_WARN("Parent pid does not match\n");
+		put_pid(p);
+		goto skip_kill;
+	}
+
+	ret = kill_pid(p, SIGTERM, 1);
+	if (ret) {
+		DRM_WARN("Failed to terminate SCU pid %d.  Performing SIGKILL.\n",core->sc_pid);
+		kill_pid(p, SIGKILL, 1);
+	}
+	udelay(1);
+	put_pid(p);
```

Comment:
Why kill SCU process here? The init function didn't launch SCU process.

I see there is a "configure_soft_kernel(info->cu_idx,info->kname,info->uuid);" in scu_probe() in scu.c. 
Why not create a "unconfigure_soft_kernel(info->cu_idx, ...);" in scu_remove() in scu.c?

Killing SKD created SCU process here looks like a bad idea to me.

Reply:
xrt_cu_scu_fini is called inside scu_remove().
The idea here is to send sigterm to the SCU process to interrupt and then cleanup the process when we need to remove SCU subdivice.

configure_soft_kernel actually only send the ioctl command to skd to start the process.  
skd then forks off the process and calls another ioctl to set the pid for the process.

Reply:
We can optimize it later but probably best to have this fix in for testing purpose first.